analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -65,12 +65,18 @@ private static ProgramState ProcessArgument(ProgramState state, IArgumentOperati
         }
 
         private static ProgramState ProcessArgumentAttributes(ProgramState state, IArgumentOperationWrapper argument, ImmutableArray<AttributeData> attributes) =>
-            attributes.Any(IsValidatedNotNullAttribute) && argument.Value.TrackedSymbol() is { } symbol
+            attributes.Any(IsNotNullAttribute) && argument.Value.TrackedSymbol() is { } symbol
                 ? state.SetSymbolConstraint(symbol, ObjectConstraint.NotNull)
                 : state;
 
-        // Same as [NotNull] https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#postconditions-maybenull-and-notnull
-        private static bool IsValidatedNotNullAttribute(AttributeData attribute) =>
-            attribute.AttributeClass?.Name == "ValidatedNotNullAttribute";
+        // https://docs.microsoft.com/dotnet/api/microsoft.validatednotnullattribute
+        // https://docs.microsoft.com/dotnet/csharp/language-reference/attributes/nullable-analysis#postconditions-maybenull-and-notnull
+        // https://www.jetbrains.com/help/resharper/Reference__Code_Annotation_Attributes.html#NotNullAttribute
+        private static bool IsNotNullAttribute(AttributeData attribute) =>
+            attribute.AttributeClass?.Name
+                is "ValidatedNotNullAttribute"
+                or "ValidatedNotNull"
```

Comment:
`Invocation_NotNullAttributeVariations` fails if I remove the line. We need a strategy on how we want to support nullable attributes. We could either check for 

* name only and *Attribute* suffix is optional
* name only and must have the *Attribute* suffix
* fully qualified name (including namespace) and must have the *Attribute* suffix (optional suffix doesn't make sense here IMO)

The first two allow for more flexibility. For `ValidatedNotNull` we can not check the namespace as it would violate the example in the rspec for S2259. We may consider being equally gracious for the attributes from the `System.Diagnostics.CodeAnalysis` and the `JetBrains.Anntoations` namespaces.

See also https://github.com/SonarSource/sonar-dotnet/pull/6120#pullrequestreview-1115321473 for throw helpers attributes.

Reply:
That's weird. `AttributeClass` is `INamedTypeSymbol`, isn't it? So it should have the real class name, no matter if the shorthand syntax was used for decoration

Reply:
The JetBrains thing is not visible from comments (like the Microsoft docs comment) nor from the UTs. 
Let's add both to be clear about it

Reply:
It's the test scaffolding that messes up the result. Because there's
```
public sealed class {attributeName} : Attribute {{ }}
```
The class itself should have the `Attribute` suffix everytime. While the usage doesn't need it. After that, it will make sense.

Reply:
We still need a strategy for what we want to do about attributes defined by the user in it its own code base. For `ValidatedNotNull` we currently support any attribute defined anywhere, that can be resolved to `[ValidatedNotNull]`. That is `class ValidatedNotNullAttribute: Attribute` as well as `class ValidatedNotNull: Attribute`. In the "Sonar" codebase, only the first variant was supported (which I think is inconsequent).
For the annotation attributes by JetBrains and .Net we need to agree whether we want to be strict (same namespace and with the "Attribute" suffix) or whether we want to relax the requirement and allow user-defined attributes of the same name in any namespace. If we allow the latter, we should also allow any attribute that can be resolved to `[NotNull]` (that is any `class NotNull` and `class NotNullAttribute` in any namespace).

Reply:
We could also add links to https://www.nuget.org/packages/Nullable and https://www.nuget.org/packages/JetBrains.Annotations in the RSpec so users don't have to redeclare the attributes themselves.

Reply:
I'm fine with name check in any namespace. As long as we have the comment to MS and JetBrains variant and we indicate so in the UT - to move the attribute into namespace and demonstrate, that different namespaces can be used. Let's say parametrized namespace for 1x MS, 1x JetBrains, 1x something totally random

Reply:
Alright. I started to follow that convention in #6120. I will wait until we have merged #6120 and the final design is settled.